wlauto/resource_getters/standard.py

```diff
@@ -522,3 +540,16 @@ def get_owner_path(resource):
         return os.path.join(os.path.dirname(__base_filepath), 'common')
     else:
         return os.path.dirname(sys.modules[resource.owner.__module__].__file__)
+
+
+def get_abi_specific_path(resource, base_dir, extension, enforce=False):
+    # Note: Using device.get_abi() method instead of the property device.abi property
+    # The property is not yet set or available at this point in the execution flow
+    abi_dir = os.path.join(base_dir, extension, resource.owner.device.get_abi())
```

Comment:
This should not be assuming that resource's owner will have a "device". The abi (or more generally, "platform") should be set in the `resource` description object (as is done for executable binaries).